### PR TITLE
post/multi/gather/ssh_creds should verify it has access to a file bef…

### LINKED ISSUE #12609: post/multi/gather/ssh_creds should verify it has access to a file before reading it
## Steps to reproduce
- if you run module post/multi/gather/ssh_creds as unprivileged user, you might encounter following exception(s) during looting

[*] Looting 3 directories
[+] Downloaded /home/fox/.ssh/authorized_keys -> /root/.msf4/loot/20191119063443_default_10.28.175.104_ssh.authorized_k_668127.txt
[-] Could not load SSH Key: Neither PUB key nor PRIV key
[-] Post failed: Rex::Post::Meterpreter::RequestError stdapi_fs_ls: Operation failed: 1
[-] Call stack:
[-]   /usr/share/metasploit-framework/lib/rex/post/meterpreter/extensions/stdapi/fs/dir.rb:62:in `entries'
[-]   /usr/share/metasploit-framework/modules/post/multi/gather/ssh_creds.rb:47:in `block in download_loot'
[-]   /usr/share/metasploit-framework/modules/post/multi/gather/ssh_creds.rb:43:in `each'
[-]   /usr/share/metasploit-framework/modules/post/multi/gather/ssh_creds.rb:43:in `download_loot'
[-]   /usr/share/metasploit-framework/modules/post/multi/gather/ssh_creds.rb:38:in `run'

## Expected behavior
Should not thrown an exception.
Read permission should be verified before accessing file.

### Metasploit version
5.0.59-dev

### I installed Metasploit with:
- [x] Kali package via apt

### OS
What OS are you running Metasploit on?
Kali 2019.4

### PR DESCRIPTION
**Pull request by MangyCoyote**
Resolves #12609 .

### CODE REVIEW

File: lib/msf/core/post/file.rb [hunk 1]
```diff
@@ -153,6 +153,19 @@ def setuid?(path)
...
   end
 
+  #
+  # See if +path+ on the remote system exists and is executable
+  #
+  # @param path [String] Remote path to check
+  #
+  # @return [Boolean] true if +path+ exists and is executable
+  #
+  def executable?(path)
```
Comment (wvu-r7) [inline lib/msf/core/post/file.rb]: Very nice. I was hoping you'd add this method and use `readable?` when I saw #12611. I don't have to request anything now. :)

**Review by wvu — APPROVED** [review]
This looks good to me.

**Review by powertothepeople2017 — APPROVED** [review]

**Comment by wvu-r7** [comment]
```
msf5 post(multi/gather/ssh_creds) > run

[*] Finding .ssh directories
[*] Looting 2 .ssh directories
[*] Looting /home/ubuntu/.ssh directory
[!] Cannot access directory: /home/ubuntu/.ssh . Missing execute permission. Skipping.
[*] Looting /home/vagrant/.ssh directory
[!] Cannot read file: /home/vagrant/.ssh/authorized_keys . Missing read permission. Skipping.
[*] Post module execution completed
msf5 post(multi/gather/ssh_creds) > run

[*] Finding .ssh directories
[*] Looting 2 .ssh directories
[*] Looting /home/ubuntu/.ssh directory
[!] Cannot access directory: /home/ubuntu/.ssh . Missing execute permission. Skipping.
[*] Looting /home/vagrant/.ssh directory
[+] Downloaded /home/vagrant/.ssh/authorized_keys -> /Users/wvu/.msf4/loot/20191209200515_default_172.28.128.5_ssh.authorized_k_729103.txt
[*] Post module execution completed
msf5 post(multi/gather/ssh_creds) >
```

https://github.com/rapid7/metasploit-framework/commit/98c586b76e5c6fdef09a2dd1a4fb5674179ee655

Thanks!

**Comment by wvu-r7** [comment]
# Release Notes

This fixes the `post/multi/gather/ssh_creds` module to check if directories have execute permission and if files have read permission.

**Pull request merged**